dex/interactor/src/structs.rs

```diff
@@ -0,0 +1,219 @@
+
+impl<M: ManagedTypeApi>
+    From<MultiValue3<EsdtTokenPayment<M>, EsdtTokenPayment<M>, EsdtTokenPayment<M>>>
+    for InteractorAddLiquidityResultType
+{
+    fn from(
+        value: MultiValue3<EsdtTokenPayment<M>, EsdtTokenPayment<M>, EsdtTokenPayment<M>>,
+    ) -> Self {
+        let extracted = value.0;
+
+        InteractorMultiValue3::new(
+            InteractorToken::from(extracted.0),
+            InteractorToken::from(extracted.1),
+            InteractorToken::from(extracted.2),
+        )
+    }
+}
+
+pub type RustBigInt = num_bigint::BigInt;
+
+#[allow(dead_code)]
+pub struct InteractorUnstakeResult {
+    pub other_token_payment: InteractorToken,
+    pub lp_farm_rewards: InteractorToken,
+    pub staking_rewards: InteractorToken,
+    pub unbond_staking_farm_token: InteractorToken,
+}
+
+impl<M: ManagedTypeApi> From<UnstakeResult<M>> for InteractorUnstakeResult {
+    fn from(value: UnstakeResult<M>) -> Self {
+        InteractorUnstakeResult {
+            other_token_payment: InteractorToken::from(value.other_token_payment),
+            lp_farm_rewards: InteractorToken::from(value.lp_farm_rewards),
+            staking_rewards: InteractorToken::from(value.staking_rewards),
+            unbond_staking_farm_token: InteractorToken::from(value.unbond_staking_farm_token),
+        }
+    }
+}
+
+#[allow(dead_code)]
+pub struct InteractorStakeProxyResult {
+    pub dual_yield_tokens: InteractorToken,
+    pub staking_boosted_rewards: InteractorToken,
+    pub lp_farm_boosted_rewards: InteractorToken,
+}
+
+impl<M: ManagedTypeApi> From<StakeProxyResult<M>> for InteractorStakeProxyResult {
+    fn from(value: StakeProxyResult<M>) -> Self {
+        InteractorStakeProxyResult {
+            dual_yield_tokens: InteractorToken::from(value.dual_yield_tokens),
+            staking_boosted_rewards: InteractorToken::from(value.staking_boosted_rewards),
+            lp_farm_boosted_rewards: InteractorToken::from(value.lp_farm_boosted_rewards),
+        }
+    }
+}
+
+#[allow(dead_code)]
+pub struct InteractorClaimDualYieldResult {
+    pub lp_farm_rewards: InteractorToken,
+    pub staking_farm_rewards: InteractorToken,
+    pub new_dual_yield_tokens: InteractorToken,
+}
+
+impl<M: ManagedTypeApi> From<ClaimDualYieldResult<M>> for InteractorClaimDualYieldResult {
+    fn from(value: ClaimDualYieldResult<M>) -> Self {
+        InteractorClaimDualYieldResult {
+            lp_farm_rewards: InteractorToken::from(value.lp_farm_rewards),
+            staking_farm_rewards: InteractorToken::from(value.staking_farm_rewards),
+            new_dual_yield_tokens: InteractorToken::from(value.new_dual_yield_tokens),
+        }
+    }
+}
+
+pub struct InteractorFarmTokenAttributes {
+    pub reward_per_share: RustBigUint,
+    pub entering_epoch: u64,
+    pub compounded_reward: RustBigUint,
+    pub current_farm_amount: RustBigUint,
+    pub original_owner: Address,
+}
+
+#[derive(Debug)]
+pub struct InteractorToken {
```

Comment:
Better name would be `InteractorPayment`, the token is the first field

Reply:
Updated.